### DISCUSSION TITLE
Testing the graphql-auth example

### DISCUSSION THREAD

**Discussion by leo-petrucci** [prisma/prisma#3650] [Q&A] [answered]
I've been working off the [typescript graphql-auth](https://github.com/prisma/prisma-examples/tree/latest/typescript/graphql-auth) example but I'm fairly confused on how to introduce automated testing to it.

I've found examples such as [this one](https://nexusjs.org/getting-started/tutorial/chapter-testing-with-prisma) that explain how to test with Nexus, but I can't seem to get them to work in the example.

How would I go about writing integration tests for my API?

**Comment by leo-petrucci** [comment, marked as answer]
If anyone stumbles upon this, I've managed to get fairly solid testing using mocha and chai. I've made a `/test/test.spec.ts` file which contains something like this:

```
export {}
const chai = require('chai')
const chaiHttp = require('chai-http')
import { GraphQLServer } from 'graphql-yoga'
const { server } = require('../src/server.ts')
const gql = require('graphql-tag')

let app: GraphQLServer

const { expect } = chai
chai.use(chaiHttp)

before(async () => {
  app = await server
})


describe('Tests', () => {
  it("test", async () => {
    const res = await chai
      .request(app)
      .post('/')
      .set('Authorization', process.env.USER_TOKEN)
      .send({
        query: `mutation { someMutation(parameters) { user { name} } }`,
      })
    expect(res.body.errors).to.be.an('array')
    expect(res.body.errors[0].message).to.be.a('string', 'Not Authorised!')
  })
})
```

Currently using `ts-mocha` to test Typescript, but the same can be done with just Mocha and js tests.

I'd love to submit a pull request to update some of the examples if anyone from Prisma would like. Let me know.

**Reply by ryands17** [reply]
@creativiii 👋 
You can also go for something like [this](https://github.com/ryands17/nexus-framework-auth) that performs the tests with Jest and Nexus framework. This is taken directly from the testing doc that you have specified above.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
